### ISSUE TITLE
Building with GCC9 fails for v5.22

### ISSUE THREAD

**Issue by rgugliel** [Project-OSRM/osrm-backend#5639]
I tried to build osrm-backend with GCC9 (which is shipped with recent version of Ubuntu).
It fails because of these lines:

- https://github.com/Project-OSRM/osrm-backend/blob/5.22/include/updater/csv_file_parser.hpp#L125
- https://github.com/Project-OSRM/osrm-backend/blob/5.22/src/server/api/parameters_parser.cpp#L50

With the error
```
src/updater/csv_source.cpp:43:31:   required from here
include/updater/csv_file_parser.hpp:125:36: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
  125 |             return std::move(result);
	  |                                    ^
include/updater/csv_file_parser.hpp:125:36: note: remove ‘std::move’ call
```

I see that the two std::move that create the issue are removed on master. Any plan to release a 5.23 soon ?

**Comment by NicolasGrosjean** [comment]
In #5511 someone has suggested to come back to GCC8.

Otherwise with @havetc we have fixed it by editing the file _CMakeLists.txt_ at the root of the repository.

We have added` -Wno-error=pessimizing-move -Wno-error=redundant-move` to the `OSRM_WARNING_FLAGS` variables.

In version 5.22.0, the line in CMakeLists.txt has been changed to
`set(OSRM_WARNING_FLAGS "-Werror=all -Werror=extra  -Werror=uninitialized -Werror=unreachable-code -Werror=unused-variable -Werror=unreachable-code -Wno-error=cpp -Wno-error=pessimizing-move -Wno-error=redundant-move -Wpedantic")`

**Comment by jcoupey** [comment]
For what it's worth, I've built `v5.22.0`  (and some previous versions) using gcc9 by applying this simple patch:

```patch
diff --git a/include/updater/csv_file_parser.hpp b/include/updater/csv_file_parser.hpp
index f3221672b..e0c7fabc0 100644
--- a/include/updater/csv_file_parser.hpp
+++ b/include/updater/csv_file_parser.hpp
@@ -122,7 +122,7 @@ template <typename Key, typename Value> struct CSVFilesParser
 
             util::Log() << "Loaded " << filename << " with " << result.size() << "values";
 
-            return std::move(result);
+            return result;
         }
         catch (const boost::exception &e)
         {
diff --git a/src/server/api/parameters_parser.cpp b/src/server/api/parameters_parser.cpp
index a2c6767bd..a68b2abcd 100644
--- a/src/server/api/parameters_parser.cpp
+++ b/src/server/api/parameters_parser.cpp
@@ -47,7 +47,7 @@ boost::optional<ParameterT> parseParameters(std::string::iterator &iter,
 
         // return move(a.b) is needed to move b out of a and then return the rvalue by implicit move
         if (ok && iter == end)
-            return std::move(parameters);
+            return parameters;
     }
     catch (const qi::expectation_failure<It> &failure)
     {
diff --git a/src/storage/io_config.cpp b/src/storage/io_config.cpp
index 82d8a914e..1e8bdf658 100644
--- a/src/storage/io_config.cpp
+++ b/src/storage/io_config.cpp
@@ -15,7 +15,7 @@ bool IOConfig::IsValid() const
     bool success = true;
     for (auto &fileName : required_input_files)
     {
-        if (!boost::filesystem::is_regular_file({base_path.string() + fileName.string()}))
+        if (!boost::filesystem::is_regular_file(boost::filesystem::path(base_path.string() + fileName.string())))
         {
             util::Log(logWARNING) << "Missing/Broken File: " << base_path.string()
                                   << fileName.string();
```

**Comment by jcoupey** [comment]
I still had to add the above patch recently when using GCC9, but from https://github.com/Project-OSRM/osrm-backend/issues/5890#issuecomment-766758154, it looks like this has been fixed in `master` after `v5.22`.

As we probably won't go through the trouble of patching `v5.22` to support GCC9 out of the box, we should probably close this? @systemed what do you think?

**Comment by systemed** [comment]
Yep - current master builds, and v5.23 has been released, so I think we can close this. Feel free to reopen if master breaks.
